website/docs/concepts/immutability.mdx

```diff
@@ -74,7 +86,7 @@ This enforces a separation of concerns and keeps business logic outside of your
 
 Here is an example of a simple immutable settings class for changing an app theme.
 
-<AutoSnippet language="dart" {...whyImmutability}></AutoSnippet>
+<AutoSnippet language="dart" {...immutability}></AutoSnippet>
```

Comment:
The snippet you created seems unrelated to the context around it. The example isn't "changing an app theme".

Also, `whyImmutability` seems unused now.

Reply:
Hey rrousselGit 👋, I'm finally circling back to this.

Yes, that's a result of me changing the page title from "Why Immutability" to just "Immutability" in [this commit](https://github.com/rrousselGit/riverpod/pull/2526/commits/534116a05b10eab4e7276ed95437bf14b1daa6c7). I didn't actually add a snippet; I just changed file names to coincide with the new page title. I figured it made more sense to change the page title since we were adding a section called "Why Immutability" to the "Why Immutability" page. I'm open to suggestions for a different title, for example—"Immutability vs. Mutability".

Even if we wanted a page title that's entirely different from Why Immutability or Immutability, I still think changing the "file name" to `immutability` in particular is nice since it's more concise and would still be appropriate for any future page titles because it directly represents the topic being discussed. However, you could argue that it diverges slightly from the current file naming convention.

Reply:
@rrousselGit, just checking in to see what you think of that change.

Reply:
Fair enough for the variable name, but my point is that the content of the code-snippet is unrelated to the description of the code snippet.

The line above  is:

> Here is an example of a simple immutable settings class for changing an app theme.

Yet the code-snippet has nothing to do with app theme, and instead deals with fruits.

Reply:
Oooh wait no, I think I got confused.

Because since you also added a new snippet under the "why_immutability.dart" name, I thought that's what this diff was about.

Reply:
Right, it's looking good on the preview. My bad.

But then, where is the fruit snippet used?

Reply:
It looks like I left it out for the moment. I assume it was because I wanted to get your approval of the code before I wired it in to the page. I still have to address some of your comments related to it, such as this one: https://github.com/rrousselGit/riverpod/pull/2526#discussion_r1189644782